airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationWriter.kt

```diff
@@ -19,6 +20,10 @@ import javax.inject.Singleton
 @Singleton
 class MockDestinationWriter : DestinationWriter {
     override fun createStreamLoader(stream: DestinationStream): StreamLoader {
+        MockDestinationBackend.deleteOldRecords(
```

Comment:
I think this is idiomatic, but lmk if I'm supposed to emitting a batch or something